feature-nft-impl/src/main/java/io/novafoundation/nova/feature_nft_impl/data/source/providers/unique_network/UniqueNetworkProvider.kt

```diff
@@ -0,0 +1,124 @@
+import io.novafoundation.nova.feature_account_api.domain.interfaces.AccountRepository
+import io.novafoundation.nova.feature_account_api.domain.model.MetaAccount
+import io.novafoundation.nova.feature_account_api.domain.model.addressIn
+import io.novafoundation.nova.feature_account_api.domain.model.requireAccountIdIn
+import io.novafoundation.nova.feature_nft_api.data.model.Nft
+import io.novafoundation.nova.feature_nft_api.data.model.NftDetails
+import io.novafoundation.nova.feature_nft_impl.data.mappers.nftIssuance
+import io.novafoundation.nova.feature_nft_impl.data.mappers.nftPrice
+import io.novafoundation.nova.feature_nft_impl.data.source.NftProvider
+import io.novafoundation.nova.feature_nft_impl.data.source.providers.unique_network.network.UniqueNetworkApi
+import io.novafoundation.nova.feature_nft_impl.data.source.providers.unique_network.network.response.UniqueNetworkNft
+import io.novafoundation.nova.runtime.multiNetwork.ChainRegistry
+import io.novafoundation.nova.runtime.multiNetwork.chain.model.Chain
+import kotlinx.coroutines.flow.Flow
+
+class UniqueNetworkNftProvider(
+    private val uniqueNetworkApi: UniqueNetworkApi,
+    private val nftDao: NftDao,
+    private val accountRepository: AccountRepository,
+    private val chainRegistry: ChainRegistry,
+) : NftProvider {
+    override val requireFullChainSync: Boolean = false
+
+    override suspend fun initialNftsSync(chain: Chain, metaAccount: MetaAccount, forceOverwrite: Boolean) {
+        val owner = metaAccount.addressIn(chain) ?: return
+        val pageSize = 100
+        var offset = 0
+        val allNfts = mutableListOf<NftLocal>()
+
+        while (true) {
+            val page = uniqueNetworkApi.getNftsPage(
+                owner = owner,
+                offset = offset,
+                limit = pageSize
+            )
+
+            if (page.items.isEmpty()) break
+
+            val pageNfts = page.items.map { remote ->
+                NftLocal(
+                    identifier = nftIdentifier(chain, remote),
+                    metaId = metaAccount.id,
+                    chainId = chain.id,
+                    collectionId = remote.collectionId.toString(),
+                    instanceId = remote.key,
+                    metadata = null,
+                    type = NftLocal.Type.UNIQUE_NETWORK,
+                    wholeDetailsLoaded = true,
+                    name = remote.name,
+                    label = "#${remote.tokenId}",
+                    media = remote.image,
+                    issuanceType = NftLocal.IssuanceType.UNLIMITED,
+                    issuanceTotal = null,
+                    issuanceMyEdition = remote.tokenId.toString(),
+                    issuanceMyAmount = null,
+                    price = null,
+                    pricedUnits = null
+                )
+            }
+
+            allNfts += pageNfts
+            offset += pageSize
+
+            if (allNfts.size >= page.count) break
```

Comment:
Does `page.count` here refer to a total number of NFTs of the user?

Reply:
Yes, page.count refers to the total number of entities matching the query.
In this case, it indicates the total number of NFTs owned by the address on the network.